bsonutil/indexes.go

```diff
@@ -33,10 +34,54 @@ var validIndexOptions = map[string]bool{
 	"wildcardProjection":      true,
 }
 
+const epsilon = 1e-9
+
+func IsIndexKeysEqual(indexKey1 bson.D, indexKey2 bson.D) bool {
+	if len(indexKey1) != len(indexKey2) {
+		// two indexes have different number of keys
+		return false
+	}
+
+	for j, elem := range indexKey1 {
+		if elem.Key != indexKey2[j].Key {
+			return false
+		}
+
+		// After ConvertLegacyIndexKeys, index key value should only be numerical or string value
+		switch key1Value := elem.Value.(type) {
+		case string:
```

Comment:
We don't need to check strings. The string "1", or in general "<number>", is impossible.

Reply:
ah, OK. If string values other than empty string is not possible, I can remove the logic which handles it.